src/shared/mkfs-util.c

```diff
@@ -101,12 +102,25 @@ int make_filesystem(
         assert(fstype);
         assert(label);
 
+        if (root && !streq(fstype, "squashfs"))
+                return log_error_errno(SYNTHETIC_ERRNO(EOPNOTSUPP),
+                                       "Populating with source tree is only supported for squashfs");
+
         if (streq(fstype, "swap")) {
                 r = find_executable("mkswap", &mkfs);
```

Comment:
or in other words, add an `if (root) return log_error_errno(SYNTHETIC_ERRNO(EINVAL), …` before this line